### ISSUE TITLE
 The expanded size of the tensor (90) must match the existing size (126) at non-singleton dimension 1.  

### ISSUE THREAD

**Issue by vcjob** [mozilla/TTS#287]
Hello everyone. 

I updated today TTS (master branch) and faced with multiple issues.
The first one, I had to add 
_"gradual_training": [[0, 7, 32], [10000, 5, 32], [50000, 3, 32], [130000, 2, 16], [290000, 1, 8]],
_ to the config_tacotron_gst.json, since it wasn't presented there, but train.py requires it (Tacotron_GST).  
The second one, which I've got stuck on, I keep facing the error: 
_RuntimeError: The expanded size of the tensor (90) must match the existing size (126) at non-singleton dimension 1.  Target sizes: [32, 90, 80].  Tensor sizes: [32, 126, 1]
_
It happens right after the first batch is fed.  Depending on data (different ds, use or no trim_silence) values differ (e.g.  _The expanded size of the tensor (100) must match the existing size (140) at non-singleton dimension 1.  Target sizes: [32, 100, 80].  Tensor sizes: [32, 140, 1]_ )
If I change model type to Tacotron2, I get the following: _AttributeError: 'Decoder' object has no attribute 'set_r'_ .
**it works with Tacotron model! (config.json)**
The full config I use is attached below. 
some specs: 
Ubuntu 18.04 
rtx2080 ti

Please, help me to make it work! Thank you!
-----------
{
        "run_name": "mozilla-tacotron-gst",
        "run_description": "GST with single speaker",

        "audio":{
            // Audio processing parameters
            "num_mels": 80,         // size of the mel spec frame.
            "num_freq": 1025,       // number of stft frequency levels. Size of the linear spectogram frame.
            "sample_rate": 22050,   // DATASET-RELATED: wav sample-rate. If different than the original data, it is resampled.
            "frame_length_ms": 50,  // stft window length in ms.
            "frame_shift_ms": 12.5, // stft window hop-lengh in ms.
            "preemphasis": 0.98,    // pre-emphasis to reduce spec noise and make it more structured. If 0.0, no -pre-emphasis.
            "min_level_db": -100,   // normalization range
            "ref_level_db": 20,     // reference level db, theoretically 20db is the sound of air.
            "power": 1.5,           // value to sharpen wav signals after GL algorithm.
            "griffin_lim_iters": 60,// #griffin-lim iterations. 30-60 is a good range. Larger the value, slower the generation.
            // Normalization parameters
            "signal_norm": true,    // normalize the spec values in range [0, 1]
            "symmetric_norm": false, // move normalization to range [-1, 1]
            "max_norm": 1,          // scale normalization to range [-max_norm, max_norm] or [0, max_norm]
            "clip_norm": true,      // clip normalized values into the range.
            "mel_fmin": 0.0,         // minimum freq level for mel-spec. ~50 for male and ~95 for female voices. Tune for dataset!!
            "mel_fmax": 10000.0,        // maximum freq level for mel-spec. Tune for dataset!!
            "do_trim_silence": false  // enable trimming of slience of audio as you load it. LJspeech (false), TWEB (false), Nancy (true)
        },

        "distributed":{
            "backend": "nccl",
            "url": "tcp:\/\/localhost:54321"
        },

        "reinit_layers": [],

        "model": "TacotronGST",          // one of the model in models/
        "grad_clip": 1,                // upper limit for gradients for clipping.
        "epochs": 10000,                // total number of epochs to train.
        "lr": 0.0001,                  // Initial learning rate. If Noam decay is active, maximum learning rate.
        "lr_decay": false,             // if true, Noam learning rate decaying is applied through training.
        "warmup_steps": 4000,          // Noam decay steps to increase the learning rate from 0 to "lr"
        "windowing": false,            // Enables attention windowing. Used only in eval mode.
        "memory_size": 5,              // ONLY TACOTRON - memory queue size used to queue network predictions to feed autoregressive connection. Useful if r < 5.
        "attention_norm": "sigmoid",   // softmax or sigmoid. Suggested to use softmax for Tacotron2 and sigmoid for Tacotron.
        "prenet_type": "original",           // "original" or "bn".
        "prenet_dropout": true,        // enable/disable dropout at prenet.
        "use_forward_attn": true,      // enable/disable forward attention. In general, it aligns faster.
        "forward_attn_mask": false,    // Apply forward attention mask at inference to prevent bad modes. Try it if your model does not align well.
        "transition_agent": false,     // enable/disable transition agent of forward attention.
        "location_attn": false,        // enable_disable location sensitive attention. It is enabled for TACOTRON by default.
        "loss_masking": true,         // enable / disable loss masking against the sequence padding.
        "enable_eos_bos_chars": false, // enable/disable beginning of sentence and end of sentence chars.
        "stopnet": true,               // Train stopnet predicting the end of synthesis.
        "separate_stopnet": true,     // Train stopnet seperately if 'stopnet==true'. It prevents stopnet loss to influence the rest of the model. It causes a better model, but it trains SLOWER.
        "tb_model_param_stats": false,     // true, plots param stats per layer on tensorboard. Might be memory consuming, but good for debugging.

        "batch_size": 32,       // Batch size for training. Lower values than 32 might cause hard to learn attention.
        "eval_batch_size":16,
        "r": 5,                 // Number of frames to predict for step.
        "gradual_training": [[0, 7, 32], [10000, 5, 32], [50000, 3, 32], [130000, 2, 16], [290000, 1, 8]],
        "wd": 0.000001,         // Weight decay weight.
        "checkpoint": true,     // If true, it saves checkpoints per "save_step"
        "save_step": 1000,      // Number of training steps expected to save traning stats and checkpoints.
        "print_step": 10,       // Number of steps to log traning on console.
        "batch_group_size": 0,  //Number of batches to shuffle after bucketing.

        "run_eval": true,
        "test_delay_epochs": 5,  //Until attention is aligned, testing only wastes computation time.
        "test_sentences_file": null,
        "data_path": "/ssd/y/", // DATASET-RELATED: can overwritten from command argument
        "meta_file_train": "/ssd/y/train.csv",      // DATASET-RELATED: metafile for training dataloader.
        "meta_file_val": "/ssd/y/val.csv",    // DATASET-RELATED: metafile for evaluation dataloader.
        "dataset": "ljspeech",      // DATASET-RELATED: one of TTS.dataset.preprocessors depending on your target dataset. Use "tts_cache" for pre-computed dataset by extract_features.py
        "min_seq_len": 1,       // DATASET-RELATED: minimum text length to use in training
        "max_seq_len": 314,     // DATASET-RELATED: maximum text length
        "output_path": "/ssd/y/outputs/",      // DATASET-RELATED: output path for all training outputs.
        "num_loader_workers": 4,        // number of training data loader processes. Don't set it too big. 4-8 are good values.
        "num_val_loader_workers": 4,    // number of evaluation data loader processes.
        "phoneme_cache_path": "new_phonems",  // phoneme computation is slow, therefore, it caches results in the given folder.
        "use_phonemes": true,           // use phonemes instead of raw characters. It is suggested for better pronounciation.
        "phoneme_language": "ru",     // depending on your target language, pick one from  https://github.com/bootphon/phonemizer#languages
        "text_cleaner": "r_cleaner",
        "use_speaker_embedding": false, // whether to use additional embeddings for separate speakers
        "style_wav_for_test": null // path to wav for styling the inference tests when using GST
    }

**Comment by DrHaqkiem** [comment]
I got the same problem also. Is it already solved ?

**Comment by vcjob** [comment]
> I got the same problem also. Is it already solved ?

Nope, for now proceeding with a plain Tacotron.

**Comment by erogol** [comment]
new dev branch would solve the problem. Let me know the result.

**Comment by vcjob** [comment]
> new dev branch would solve the problem. Let me know the result.

It works in dev branch! Thank you! (Tacotron2; after I added _ "gradual_training": null_ in the config.

**Comment by DrHaqkiem** [comment]
> > new dev branch would solve the problem. Let me know the result.
> 
> It works in dev branch! Thank you! (Tacotron2; after I added _"gradual_training": null_ in the config.

YEAH it works for me too @erogol . THANK YOU!

Anyway at first i got memory issue with the default number of wrokers=4.
But then i change the value to 0 for both **num_load_workers** and **num_val_workers**.

What both of u think about this? @erogol @erogol  ?
